plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/query/index/HudiSecondaryIndexSupport.java

```diff
@@ -78,6 +84,7 @@ public Map<String, List<FileSlice>> lookupCandidateFilesInMetadataTable(
         // TODO: document here what this map is keyed by
         Map<String, HoodieRecordGlobalLocation> recordKeyLocationsMap = metadataTable.readSecondaryIndex(secondaryKeys, indexName);
         if (recordKeyLocationsMap.isEmpty()) {
+            timer.endTimer();
```

Comment:
nit: this is not necessary if the duration is not used, as the timer pushes the timing information to a queue only.

Reply:
Got it.  This is a minor issue as I see `timer.endTimer()` call in a few places without the time being used.